Source/WebCore/loader/cache/MemoryCache.cpp

```diff
@@ -307,7 +308,7 @@ void MemoryCache::pruneDeadResources()
     LOG(ResourceLoading, "MemoryCache::pruneDeadResources");
 
     unsigned capacity = deadCapacity();
-    if (capacity && m_deadSize <= capacity)
+    if (!m_deadSize || (capacity && m_deadSize <= capacity))
```

Comment:
We should be able to move the `m_deadSize` check to before calculating the capacity.

Same with the `m_liveSize` check.